### PR TITLE
Support longer token symbols via `wallet_watchAsset`

### PR DESCRIPTION
**Pull request by Gudahtt**
This increases the maximum token length that can be suggested via `wallet_watchAsset` from 6 to 11, which matches the maximum symbol length for custom tokens added on the MetaMask extension.

### CODE REVIEW

File: src/util.test.ts [hunk 1]
```diff
@@ -498,14 +498,14 @@ describe('util', () => {
       ).toThrow('Invalid symbol: not a string.');
     });
 
-    it('should throw if symbol is more than 6 characters long', () => {
+    it('should throw if symbol is more than 11 characters long', () => {
```
Comment (ryanml) [inline src/util.test.ts:501]: nit: could we retain this old test case using `TKNTKNTKN` as the symbol, and update the outcome to prove that an error is not thrown? then we can add a new test case for `TKNTKNTKNTKN`, just to cover both bases

Comment (Gudahtt) [reply src/util.test.ts:501]: Sure! We can do that.

Comment (Gudahtt) [reply src/util.test.ts:501]: Actually, instead I'll ensure the a symbol at the maximum length is still accepted, to ensure we're checking all boundary conditions. So length 11 rather than 6.

Comment (ryanml) [reply src/util.test.ts:501]: sgtm!

Comment (Gudahtt) [reply src/util.test.ts:531]: Additional tests added in 5ba4b25f6f047004e92c486fd092509d9eb1f0e8

**Review by rekmarks — APPROVED** [review]
LGTM!

**Review by ryanml — APPROVED** [review]
++ thanks for the changes!

**Comment by Gudahtt** [comment]
Thanks for the reviews! Marking this as "DO-NOT-MERGE" for now until the mobile team has signed off on this.

**Comment by Gudahtt** [comment]
Removing the `DO-NOT-MERGE` label, as it seems everyone on the mobile team has signed off on this.
